### DISCUSSION TITLE
Blog post on building bubbletea programs

### DISCUSSION THREAD

**Discussion by leg100** [charmbracelet/bubbletea#1114] [Show and tell]
Hello, I've put up a blog post on building bubbletea programs. It's not a tutorial but provides advice on the stages thereafter:

https://leg100.github.io/en/posts/building-bubbletea-programs/

**Comment by blvrd** [comment]
This is fantastic, thanks for writing it! I've struggled with several of the points in your post, especially the layout arithmetic.

**Comment by bashbunni** [comment]
This is outstanding! Thank you so much for producing this 🎊 🫶

**Comment by Broderick-Westrope** [comment]
I enjoyed this read. The "tree of models" is a concept that has really helped organise my past projects. Awesome job!

**Comment by fchastanet** [comment]
Thanks a lot! it should definitvely be part of bubbletea documentation
I was searching a way to easily do a focus management, I will follow your advice with the root model
